irc/client.go

```diff
@@ -1022,6 +1022,21 @@ func (client *Client) ModeString() (str string) {
 	return "+" + client.modes.String()
 }
 
+// IsFriend() returns true if the given otherClient shares a channel with this client, or if they are the same user.
+func (client *Client) IsFriend(otherClient *Client) bool {
+	if client == otherClient {
```

Comment:
@slingamn do we need to iterate over `client.Sessions()` here like we do down in Friends? returning `true` when client == otherClient definitely makes sense

Reply:
This is fine. `Friends` needs to be aware of sessions because it respects caps (which can vary across sessions). But channel membership is per client, not per session (you can't have one session in #bar and another one outside it).